File: dart_sample/lib/demos/cupertino/cupertino_navigation_bar_demo_page.dart
import 'package:cupertino_ui/cupertino_ui.dart';
import 'package:material_ui/material_ui.dart';

import '../../counter_widgets.dart';

class CupertinoNavigationBarDemoPage extends StatefulWidget {
  const CupertinoNavigationBarDemoPage({super.key});

  @override
  State<CupertinoNavigationBarDemoPage> createState() =>
      _CupertinoNavigationBarDemoPageState();
}

class _CupertinoNavigationBarDemoPageState extends State<CupertinoNavigationBarDemoPage> {
  bool _searchActive = false;

  @override
  Widget build(BuildContext context) {
    return CupertinoTheme(
      data: const CupertinoThemeData(),
      child: CupertinoPageScaffold(
        child: CustomScrollView(
          slivers: <Widget>[
            CupertinoSliverNavigationBar.search(
              searchField: const CupertinoSearchTextField(),
              largeTitle: const Text('Nav bars'),
              stretch: true,
              onSearchableBottomTap: (bool active) =>
                  setState(() => _searchActive = active),
            ),
            SliverToBoxAdapter(child: Builder(builder: _buildContent)),
          ],
        ),
      ),
    );
  }

  Widget _buildContent(BuildContext context) {
    final Color label = CupertinoDynamicColor.resolve(
      CupertinoColors.label,
      context,
    );
    final Color secondaryLabel = CupertinoDynamicColor.resolve(
      CupertinoColors.secondaryLabel,
      context,
    );

    return Padding(
      padding: const EdgeInsets.all(16.0),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        spacing: 10.0,
        children: <Widget>[
          Text(
            _searchActive
                ? 'Search is active — Cancel restores the collapsed bar.'
                : 'Scroll to collapse the large title, overscroll to stretch '
                      'it, or tap the field to search.',
            style: TextStyle(fontSize: 14.0, color: secondaryLabel),
          ),
          _buildAction(
            'Push detail page (auto middle + back label)',
            () => _pushDetail(context),
            const Color(0xFFE9F0FF),
          ),
          _buildAction(
            "Push page with a long title ('Back' fallback)",
            () => _pushLongTitle(context),
            const Color(0xFFEAE4FF),
          ),
          ...List<Widget>.generate(
            20,
            (int index) => Text(
              'Row ${index + 1}',
              style: TextStyle(fontSize: 14.0, color: label),
            ),
          ),
        ],
      ),
    );
  }

  static void _pushDetail(BuildContext context) {
    Navigator.of(context).push(
      CupertinoPageRoute<Object?>(
        title: 'Details',
        builder: (_) => CupertinoPageScaffold(
          navigationBar: const CupertinoNavigationBar(),
          child: Center(
            child: Builder(
              builder: (BuildContext routeContext) => Text(
                'The static bar implies its middle title and the back label '
                'from the route titles.',
                textAlign: TextAlign.center,
                style: TextStyle(
                  fontSize: 14.0,
                  color: CupertinoDynamicColor.resolve(
                    CupertinoColors.label,
                    routeContext,
                  ),
                ),
              ),
            ),
          ),
        ),
      ),
    );
  }

  static void _pushLongTitle(BuildContext context) {
    Navigator.of(context).push(
      CupertinoPageRoute<Object?>(
        title: 'Extended configuration options',
        builder: (_) => CupertinoPageScaffold(
          child: CustomScrollView(
            slivers: <Widget>[
              const CupertinoSliverNavigationBar(),
              SliverToBoxAdapter(
                child: Padding(
                  padding: const EdgeInsets.all(16.0),
                  child: Builder(
                    builder: (BuildContext routeContext) => Column(
                      crossAxisAlignment: CrossAxisAlignment.stretch,
                      spacing: 10.0,
                      children: <Widget>[
                        Text(
                          'This title is over 12 characters, so the next '
                          "page's back label falls back to 'Back'.",
                          style: TextStyle(
                            fontSize: 14.0,
                            color: CupertinoDynamicColor.resolve(
                              CupertinoColors.label,
                              routeContext,
                            ),
                          ),
                        ),
                        _buildAction(
                          "Push detail from here (shows 'Back')",
                          () => _pushDetail(routeContext),
                          const Color(0xFFE8F4E8),
                        ),
                      ],
                    ),
                  ),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }

  static Widget _buildAction(String label, VoidCallback onTap, Color background) {
    return CounterTapButton(
      label: label,
      onTap: onTap,
      background: background,
      foreground: Colors.black,
      fontSize: 12.0,
      padding: const EdgeInsets.symmetric(horizontal: 10.0, vertical: 8.0),
    );
  }
}

File: src/Sample/Plumix.Sample/Demos/Cupertino/CupertinoNavigationBarDemoPage.cs
using System;
using System.Linq;
using Avalonia;
using Avalonia.Media;
using Plumix.Cupertino;
using Plumix.Rendering;
using Plumix.UI;
using Plumix.Widgets;

namespace Plumix;

// Dart parity source (reference): dart_sample/lib/demos/cupertino/cupertino_navigation_bar_demo_page.dart (exact sample parity)

public sealed class CupertinoNavigationBarDemoPage : StatefulWidget
{
    public override State CreateState() => new CupertinoNavigationBarDemoPageState();
}

internal sealed class CupertinoNavigationBarDemoPageState : State
{
    private bool _searchActive;

    public override Widget Build(BuildContext context)
    {
        return new CupertinoTheme(
            data: new CupertinoThemeData(),
            child: new CupertinoPageScaffold(
                child: new CustomScrollView(
                    slivers:
                    [
                        CupertinoSliverNavigationBar.Search(
                            searchField: new CupertinoSearchTextField(),
                            largeTitle: new Text("Nav bars"),
                            stretch: true,
                            onSearchableBottomTap: active => SetState(() => _searchActive = active)),
                        new SliverToBoxAdapter(new Builder(BuildContent)),
                    ])));
    }

    private Widget BuildContent(BuildContext context)
    {
        Color label = CupertinoDynamicColor.Resolve(CupertinoColors.Label, context);
        Color secondaryLabel = CupertinoDynamicColor.Resolve(CupertinoColors.SecondaryLabel, context);

        return new Padding(
            new Thickness(16.0),
            child: new Column(
                crossAxisAlignment: CrossAxisAlignment.Stretch,
                spacing: 10.0,
                children:
                [
                    new Text(
                        _searchActive
                            ? "Search is active — Cancel restores the collapsed bar."
                            : "Scroll to collapse the large title, overscroll to stretch it, "
                              + "or tap the field to search.",
                        fontSize: 14.0,
                        color: secondaryLabel),
                    BuildAction(
                        "Push detail page (auto middle + back label)",
                        () => PushDetail(context),
                        Color.Parse("#FFE9F0FF")),
                    BuildAction(
                        "Push page with a long title ('Back' fallback)",
                        () => PushLongTitle(context),
                        Color.Parse("#FFEAE4FF")),
                    .. Enumerable.Range(1, 20)
                        .Select(index => (Widget)new Text($"Row {index}", fontSize: 14.0, color: label)),
                ]));
    }

    private static void PushDetail(BuildContext context)
    {
        Navigator.Of(context).Push(new CupertinoPageRoute<object?>(
            title: "Details",
            builder: _ => new CupertinoPageScaffold(
                navigationBar: new CupertinoNavigationBar(),
                child: new Center(
                    child: new Builder(routeContext => new Text(
                        "The static bar implies its middle title and the back label "
                        + "from the route titles.",
                        fontSize: 14.0,
                        textAlign: TextAlign.Center,
                        color: CupertinoDynamicColor.Resolve(CupertinoColors.Label, routeContext)))))));
    }

    private static void PushLongTitle(BuildContext context)
    {
        Navigator.Of(context).Push(new CupertinoPageRoute<object?>(
            title: "Extended configuration options",
            builder: _ => new CupertinoPageScaffold(
                child: new CustomScrollView(
                    slivers:
                    [
                        new CupertinoSliverNavigationBar(),
                        new SliverToBoxAdapter(new Padding(
                            new Thickness(16.0),
                            child: new Builder(routeContext => new Column(
                                crossAxisAlignment: CrossAxisAlignment.Stretch,
                                spacing: 10.0,
                                children:
                                [
                                    new Text(
                                        "This title is over 12 characters, so the next page's "
                                        + "back label falls back to 'Back'.",
                                        fontSize: 14.0,
                                        color: CupertinoDynamicColor.Resolve(
                                            CupertinoColors.Label,
                                            routeContext)),
                                    BuildAction(
                                        "Push detail from here (shows 'Back')",
                                        () => PushDetail(routeContext),
                                        Color.Parse("#FFE8F4E8")),
                                ])))),
                    ]))));
    }

    private static Widget BuildAction(string label, Action onTap, Color background)
    {
        return new CounterTapButton(
            label: label,
            onTap: onTap,
            background: background,
            foreground: Colors.Black,
            fontSize: 12.0,
            padding: new Thickness(10.0, 8.0));
    }
}
